### PR TITLE
Giveaway system

### LINKED ISSUE #67: [Enhancement]: Giveaway command
### Explain your idea.

Adding a giveaway command that allows you to create a giveaway that takes the following arguments (either through slash command options and/or a modal)

- Channel to send the giveaway in
-  Text (paragraph if you do this in a modal) for the a description of the giveaway
- Number of winners allowed to win
- The amount of time for the giveaway (maybe hours in double format)

### How could you do this differently

_No response_

### Anything else?

The poll command will be fairly similar. I recommend you wait till you see my code on this to keep our designs consistent if you are planning on PRing this,

### PR DESCRIPTION
**Pull request by Zcamt**
Implementation of giveaway system

Closes #67

### CODE REVIEW

File: src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        title = "Error!"
+                        description = "The time you specified is not a number"
+                    }).queue()
+                    return@listener
+                }
+
+                if(rawTime <= 0) {
+                    it.replyEmbed({
+                        title = "Error!"
+                        description = "The time you specified is negative. We haven't built a time machine (yet!)."
```
Comment (Nuckerr) [inline src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt:51]: Where have I seen that before 😂😂

File: src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.learnspigot.bot.entity
+
+import dev.morphia.annotations.Entity
+import dev.morphia.annotations.Id
+import java.time.Instant
+
+@Entity("giveaways")
+data class Giveaway(
+    @Id val id: String,
+    val channelId: String = "",
```
Comment (Nuckerr) [inline src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt:10]: Surely this shouldn't need a default value. I think we only need default values for migrated values

File: src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt (hunk 1 above)
Comment (Zcamt) [reply src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt:51]: > Where have I seen that before 😂😂

N..no..nowhere, I promise this is all my own work

File: src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt (hunk 2 above)
Comment (Zcamt) [reply src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt:10]: > Surely this shouldn't need a default value. I think we only need default values for migrated values

If they don't have a default value findGiveaway starts yelling at me :(

File: src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt (hunk 1 above)
Comment (sttephen) [reply src/main/kotlin/com/learnspigot/bot/command/GiveawayCommand.kt:51]: Your honour.

File: src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt (hunk 2 above)
Comment (Nuckerr) [reply src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt:10]: Ah ok

**Review by Nuckerr — APPROVED** [review]

**Comment by Nuckerr** [comment]
I will test later

File: src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt (hunk 2 above)
Comment (Zcamt) [reply src/main/kotlin/com/learnspigot/bot/entity/Giveaway.kt:10]: There might and probably is a smarter way, but I couldn’t figure it out. As a precaution any “broken” giveaways are removed on startup tho so they won’t persist

**Comment by Nuckerr** [comment]
Ok, so it all works. Just a couple things:
- When I run the command, it works however I get this error: 
- The command interaction is left on thinking. Makes some sort of response
- Remove the relative timestamp when the give away ends.
- Remove the button when the give away ends

**Comment by Zcamt** [comment]
> Ok, so it all works. Just a couple things:
> 
> * When I run the command, it works however I get this error:
> * The command interaction is left on thinking. Makes some sort of response
> * Remove the relative timestamp when the give away ends.
> * Remove the button when the give away ends

1. Can't see the error you're referring to 
2. Will fix
3. Sure
4. Will fix

**Comment by Nuckerr** [comment]
Haha, forgot to attach the pastebin - http://paste.learnspigot.com/imuwodefah

**Comment by Zcamt** [comment]
> Haha, forgot to attach the pastebin - http://paste.learnspigot.com/imuwodefah

The error seems to come from the poll command, my guess is that this https://github.com/learnspigot/discord-bot/blob/d0557f0aa3ee66bfbf7234c19773fac52e262187/src/main/kotlin/com/learnspigot/bot/command/PollCommand.kt#L73

Needs to be moved up so it's inside the scope of the if-statement, else it'll acknowledge other modal events.

**Pull request merged**